### ISSUE TITLE
The data / folder structure

### ISSUE THREAD

**Issue by MalindaVania** [MASILab/EssNet#2]
Hi, I would like to train my own dataset, and I checked the data structure in sublist. But it seems like there are some missing information how to structure the folder data. Could you please share the information about it? Did you make the 3 folder for the data? MRI data, CT data and the MRI Label data?

Or maybe you can share your dummy data so I can try to debug it myself?

Thank you

**Comment by yuankaihuo** [comment]
Thanks for your questions.
You can see the subject list files, it will show you the structure that I used.
https://github.com/MASILab/SynSeg-Net/tree/master/sublist

> Hi, I would like to train my own dataset, and I checked the data structure in sublist. But it seems like there are some missing information how to structure the folder data. Could you please share the information about it? Did you make the 3 folder for the data? MRI data, CT data and the MRI Label data?
> 
> Or maybe you can share your dummy data so I can try to debug it myself?
> 
> Thank you

**Comment by MalindaVania** [comment]
Thank you for the response, I followed the structure and change a little to put the label/segmentation. But when I check the dataset number using len(dataset), there is the different number on the train.py 

line 62, 63 and 64

data_loader = CreateDataLoader(opt)
    dataset = data_loader.load_data()
    dataset_size = len(data_loader)

the dataset_size is 10 because I try to put small data.
but in the debugging process when I used len(dataset) it resulted in 3, not 10
so there is a difference for data using the command in line 63
dataset=data_loader.load_data()

with the dataset that process in data_loader in line 64

![Capture_debug](https://user-images.githubusercontent.com/26785712/55444918-c1e5fd80-55f3-11e9-88ab-9a9f6154773f.PNG)

and I think this difference raises an error that halts the training process 

![image](https://user-images.githubusercontent.com/26785712/55444970-f78ae680-55f3-11e9-94be-401b3653adb7.png)
![image](https://user-images.githubusercontent.com/26785712/55444994-20ab7700-55f4-11e9-8fe9-2a92bcdaf65a.png)

Do you have any suggestions about what might be the cause?

Thank you

**Comment by yuankaihuo** [comment]
dataset and data_loader are two different things.
It might be easier to use PyCharm or similar IDE to debug it.

> Thank you for the response, I followed the structure and change a little to put the label/segmentation. But when I check the dataset number using len(dataset), there is the different number on the train.py
> 
> line 62, 63 and 64
> 
> data_loader = CreateDataLoader(opt)
> dataset = data_loader.load_data()
> dataset_size = len(data_loader)
> 
> the dataset_size is 10 because I try to put small data.
> but in the debugging process when I used len(dataset) it resulted in 3, not 10
> so there is a difference for data using the command in line 63
> dataset=data_loader.load_data()
> 
> with the dataset that process in data_loader in line 64
> 
> ![Capture_debug](https://user-images.githubusercontent.com/26785712/55444918-c1e5fd80-55f3-11e9-88ab-9a9f6154773f.PNG)
> 
> and I think this difference raises an error that halts the training process
> 
> ![image](https://user-images.githubusercontent.com/26785712/55444970-f78ae680-55f3-11e9-94be-401b3653adb7.png)
> ![image](https://user-images.githubusercontent.com/26785712/55444994-20ab7700-55f4-11e9-8fe9-2a92bcdaf65a.png)
> 
> Do you have any suggestions about what might be the cause?
> 
> Thank you
